src/ProductView/TabBar/QueryInterface/QueryInterface.js

```diff
@@ -118,10 +122,18 @@ class QueryInterface extends Component {
     }
 
     handleChangeQuery(event) {
-        this.setState({ eventQuery: event.target.value });
+        const queryValue = event.target.value;
+        const modalIsAbleToSave = this.isModalAbleToSave(this.state.eventTypeName, queryValue);
+        this.setState({ eventQuery: queryValue, modalIsAbleToSave: modalIsAbleToSave });
+        this.eventQueryFormValidation(queryValue);
+    }
+
+    isModalAbleToSave(eventTypeName, eventQuery) {
+        return eventTypeName && eventQuery;
     }
 
     handleSaveQuery() {
+        this.setState({ modalLoading: true });
         const eventType = this.buildJSONSerializableDataContainer();
```

Comment:
This stuff should be done on DataTransmitter, because it relates to network stuff. This does not have anything to do with queries.